strings/base_fast_forward.h

```diff
@@ -112,6 +116,11 @@ namespace winrt::impl
             return self->m_owner->GetTrustLevel(level);
         }
 
+
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmicrosoft-cast"
```

Comment:
Is this just because it's converting a function pointer to `void*`? I'm curious if you can convert these to `void(*)()` - the function pointer equivalent of `void*` - and make things work equivalently. GitHub makes it a little tricky to see, but I'm _hoping_ these are in the `impl` namespace and therefore such a change won't be breaking.

Idk how much such a change might break internally, so it might not be desirable to fully explore right now, so perhaps as a follow-up.

Reply:
The warning was something along the lines of these casts only working on Microsoft toolchains.  cppwinrt can be used with different compilers, but as far as I know the outputs only ever run on Windows, so that is likely fine.  I would definitely prefer to leave this for a future checkin because I've already spent 10x longer on this PR than I expected going into it.

Reply:
> The warning was something along the lines of these casts only working on Microsoft toolchains

I'm _guessing_ that warning just means that MSVC allows the cast, not that it'll only work on Windows. But anyway, I agree that it's not necessary to address it right now.